app/com/lynxanalytics/biggraph/graph_operations/ExportToNeo4j.scala

```diff
@@ -0,0 +1,135 @@
+  extends SparkOperation[ExportAttributesToNeo4j.Input, ExportAttributesToNeo4j.Output] {
+  val neo = Neo4jConnection(url, username, password)
+  @transient override lazy val inputs = new ExportAttributesToNeo4j.Input()
+  def outputMeta(instance: MetaGraphOperationInstance) = new ExportAttributesToNeo4j.Output()(instance)
+  override def toJson = Json.obj(
+    "url" -> url, "username" -> username, "password" -> password, "labels" -> labels,
+    "keys" -> keys, "version" -> version, "nodesOrRelationships" -> nodesOrRelationships)
+  def execute(
+    inputDatas: DataSet,
+    o: ExportAttributesToNeo4j.Output,
+    output: OutputBuilder,
+    rc: RuntimeContext): Unit = {
+    implicit val ds = inputDatas
+    // Drop null keys.
+    val df = keys.foldLeft(inputs.t.df)((df, key) => df.filter(df(key).isNotNull))
+    val keyMatch = keys.map(k => s"`$k`: event.`$k`").mkString(", ")
+    val query = nodesOrRelationships match {
+      case "nodes" => s"MATCH (n$labels {$keyMatch}) SET n += event"
+      case "relationships" => s"MATCH ()-[r$labels {$keyMatch}]-() SET r += event"
+    }
+    neo.send(df, query)
+    val exportResult = "Export done."
+    output(o.exportResult, exportResult)
+  }
+}
+
+object ExportGraphToNeo4j extends OpFromJson {
+  class Input extends MagicInputSignature {
+    val vs = table
+    val es = table
+  }
+  class Output(implicit instance: MetaGraphOperationInstance) extends MagicOutput(instance) {
+    val exportResult = scalar[String]
+  }
+  def fromJson(j: JsValue) = ExportGraphToNeo4j(
+    (j \ "url").as[String], (j \ "username").as[String], (j \ "password").as[String],
+    (j \ "nodeLabelsColumn").as[String], (j \ "relationshipTypeColumn").as[String],
+    (j \ "version").as[Long])
+  val VID = "!LynxKite ID"
+  val SRCDST = "!LynxKite endpoint IDs"
+  val SRCID = "!Source LynxKite ID"
+  val DSTID = "!Destination LynxKite ID"
+}
+
+case class ExportGraphToNeo4j(
+    url: String, username: String, password: String, nodeLabelsColumn: String,
+    relationshipTypeColumn: String, version: Long)
+  extends SparkOperation[ExportGraphToNeo4j.Input, ExportGraphToNeo4j.Output] {
+  import ExportGraphToNeo4j._
+  val neo = Neo4jConnection(url, username, password)
+  @transient override lazy val inputs = new ExportGraphToNeo4j.Input()
+  def outputMeta(instance: MetaGraphOperationInstance) = new ExportGraphToNeo4j.Output()(instance)
+  override def toJson = Json.obj(
+    "url" -> url, "username" -> username, "password" -> password,
+    "nodeLabelsColumn" -> nodeLabelsColumn, "relationshipTypeColumn" -> relationshipTypeColumn,
+    "version" -> version)
+  def execute(
+    inputDatas: DataSet,
+    o: ExportGraphToNeo4j.Output,
+    output: OutputBuilder,
+    rc: RuntimeContext): Unit = {
+    implicit val ds = inputDatas
+    val F = spark.sql.functions
+    // Prefix the internal IDs with this operation's GUID so different exports don't collide.
+    val guid = F.lit(this.gUID.toString + "-")
```

Comment:
Does this make sense?

Reply:
Yeah, I think this is perfect! But what about adding GUID as vertex and edge attribute as well? That would allow users to clean up results of somehow screwed up exports with relative ease. Maybe we can add export date as well?

Reply:
> Yeah, I think this is perfect! But what about adding GUID as vertex and edge attribute as well? That would allow users to clean up results of somehow screwed up exports with relative ease. Maybe we can add export date as well?

Actually the timestamp is better than the GUID. Maybe you've recreated your LynxKite instance. Maybe two LynxKite instances use the same Neo4j instance. If everyone exports the example graph, you can end up with two exports with the same GUID quick enough.

I'll switch to just using the timestamp. Easier to interpret too.

Reply:
👍